src/app/Shared/Services/Api.service.tsx

```diff
@@ -289,6 +303,21 @@ export class ApiService {
     );
   }
 
+  createSnapshotV2(): Observable<Recording> {
```

Comment:
Might need to annotate this as `Recording|undefined` since catchError return undefined. I think right now, no error shows because resp.json() return Promise<any>, which passes all type guards.

Reply:
I think we could also define another interface `RecordingReponse` that extends `ApiV2Response` for this. Then annotation `resp.json() as RecordingReponse`. Both cases are the same as ts dont check runtime type but with this, we can have some nice type hints/guards.